src/Symfony/Component/HttpFoundation/StreamedJsonResponse.php

```diff
@@ -71,62 +72,60 @@ private function stream(): void
         $generators = [];
         $structure = $this->data;
 
-        array_walk_recursive($structure, function (&$item, $key) use (&$generators) {
+        array_walk_recursive($structure, function (&$item) use (&$generators) {
             // generators should be used but for better DX all kind of Traversable are supported
             if ($item instanceof \Traversable && !$item instanceof \JsonSerializable) {
-                // using uniqid to avoid conflict with eventually other data in the structure
-                $placeholder = uniqid('__placeholder_', true);
-                $generators[$placeholder] = $item;
-
-                $item = $placeholder;
+                $generators[] = $item;
+                $item = self::PLACEHOLDER;
+            } elseif (self::PLACEHOLDER === $item) {
+                $generators[] = $item;
             }
         });
 
         $jsonEncodingOptions = \JSON_THROW_ON_ERROR | $this->getEncodingOptions();
-        $structureText = json_encode($structure, $jsonEncodingOptions);
-
-        foreach (array_keys($generators) as $placeholder) {
-            // split structure json by placeholder for stream the before and between part of the generator items
-            [$start, $end] = explode('"'.$placeholder.'"', $structureText, 2);
-
-            // send first and between parts of the structure
-            echo $start;
-
-            $count = 0;
-            $startTag = '[';
-            foreach ($generators[$placeholder] as $key => $item) {
-                if (0 === $count) {
-                    // depending of the first elements key the generator is detected as a list or map
-                    // we can not check for a whole list or map because that would hurt the performance
-                    // of the streamed response which is the main goal of this response class
-                    if ($key !== 0) {
-                        $startTag = '{';
+        $keyEncodingOptions = $jsonEncodingOptions & ~\JSON_NUMERIC_CHECK;
```

Comment:
We need to disable `JSON_NUMERIC_CHECK` when encoding the key to be sure than we get a quoted string whatever encoding options are given.

Reply:
Make sense.